### ISSUE TITLE
LogViewer size limit configuration

### ISSUE THREAD

**Issue by RolandVExp** [rundeck/rundeck#8042]
**Is your feature request related to a problem? Please describe.**
Since Rundeck version 3.4 a limitation has been set to prevent logViewer from displaying large log outputs. I understand this has been done to prevent memory usage on server and browser side, but I think this would be great to be able to configure the maximum log size before getting the "whale of a log!" message.

**Describe the solution you'd like**
An additional setting in rundeck-config.properties that would allow us to change the value of maxLogSize defined in the logViewer.vue template.

**Additional context**
This would allow larger log files viewing if we choose to.
![whaleLog](https://user-images.githubusercontent.com/49242120/206657641-3410a338-1483-40d8-bfd0-c607b12af1e5.png)

**Comment by mshedsilegx** [comment]
I agree with the comment above. We are facing the exact same issue when our logs > 1MB (ex: 7000 lines). For example, a log of 3.5MB with 21056 lines is generating the warning in our case.

**Comment by flotho** [comment]
Hi,
I have the same "issue" and I'm wondering if couldn't cause memory issues on a LXC container.

**Comment by emperortomato** [comment]
We desperately need the ability to tune the max log size cutoff, which for our purposes we'd probably set to infinity and deal with the potential browser slowdown. The refusal to display "large" logs (3MB, large?) basically breaks our ability to troubleshoot failed steps in Rundeck. The thing that really bugs me is this is only a few dozen lines that I want to display, but the _cumulative_ log file (for all hosts + steps) has reached the cutoff threshold, so it refuses to display this handful of lines:

![image](https://github.com/rundeck/rundeck/assets/10518611/616d2ef3-d5d0-4f87-877d-48074d887e68)

Downloading the logs isn't a good option, because with hundreds of nodes simultaneously executing steps, coupled with Rundeck no longer including the node name in log messages, we're stuck looking at an unreadable jumble of log messages from every host at different points in a job. Finding the actual "failure" message for a failed step in the middle of all of that can be a challenge.

_Please_ give us the ability to tune this threshold.

**Comment by mshedsilegx** [comment]
@fdevans 
Another nice write up on the issue. As I stated, I believe it is a major limitation of the product currently.
This has been raised for years for no resolution so far. To my knowledge, no other products in that space have this limitation.

Linked issues:
#2852
#6223
#6469
#7300
#8042
and probably plenty more

**Comment by mshedsilegx** [comment]
It is confirmed also on 5.0.1, and with very small logs, ie 3-4MB, but with a lot of line items. I'm surprised that we do not get more attention from development, as this qualifies IMO as a major deficiency of the product, and has been open for years.

**Comment by mshedsilegx** [comment]
@fdevans  for update

**Comment by kahnman** [comment]
Any update on this? We're using Rundeck to run some Ansible jobs and with Ansible logs being very verbose, this log limit quickly kills our ability to see the results of the Ansible playbook.

**Comment by mshedsilegx** [comment]
Issue with maxLogSize reported in #9172

**Comment by MartinMachl** [comment]
We are running 5.3.0 and when accessing a completed job and checking the log output the log is incomplete.

The job ran until shortly after 20:00. When accessing the job log on disk the file is complete.
![Screenshot 2024-06-20 at 09 32 02](https://github.com/rundeck/rundeck/assets/70367852/3bb2e8a6-e289-4f91-a291-cc68e1aa9b64)

**Comment by MegaDrive68k** [comment]
Hi,

After testing on Rundeck 5.10, the `maxLogSize` property does work. However, Node View still truncates what it shows on screen if the log is huge (1000 lines limit), even if it's within the size limit. This is just to keep things running smoothly in the UI.

You can see the full log in the Log Output view.

Regards!

**Comment by fdevans** [comment]
Thanks for the report. As part of a cleanup of our issue tracker, we're closing issues older than 12 months, in line with Rundeck's support policy of one year from a release's date (see the [release calendar](https://docs.rundeck.com/docs/history/release-calendar.html) for currently supported versions). This isn't a judgment that the report is invalid — many of these are likely still real — but we can't verify against a version this old.

If you're still seeing this on a currently supported release (6.1.x or 5.20.x+), please open a new issue using our [bug report template](https://github.com/rundeck/rundeck/issues/new?template=bug_report.yml) with your version and reproduction steps. Please don't reopen this issue — a fresh issue keeps things tracked against a supported version and gets it in front of us properly. We're actively working on improving how issues get submitted and triaged going forward.

Thanks for your patience,
The Rundeck Team
